spec/paperclip/storage/s3_spec.rb

```diff
@@ -379,6 +379,52 @@ def counter
     end
   end
 
+  context "An attachment that uses S3 for storage and has styles" do
+    before do
+      rebuild_model(
+        (aws2_add_region).merge storage: :s3,
+        styles: { thumb: ["90x90#", :jpg] },
+        bucket: "bucket",
+        s3_credentials: { "access_key_id" => "12345",
+                          "secret_access_key" => "54321" }
+        )
+
+      @file = File.new(fixture_file("5k.png"), "rb")
+      @dummy = Dummy.new
+      @dummy.avatar = @file
+      @dummy.save
+    end
+
+    context "reprocess" do
+      before do
+        @object = stub
+        @dummy.avatar.stubs(:s3_object).with(:original).returns(@object)
+        @dummy.avatar.stubs(:s3_object).with(:thumb).returns(@object)
+        @object.stubs(defined?(::Aws) ? :get : :read).with().yields(@file.read)
+        @object.stubs(:exists?).with().returns(true)
+      end
+
+      it "uploads original" do
+        @object.expects((defined?(::Aws) ? :upload_file : :write))
+          .with(anything, content_type: 'image/png',
+                acl: Paperclip::Storage::S3::DEFAULT_PERMISSION).returns(true)
+        @object.expects((defined?(::Aws) ? :upload_file : :write))
+          .with(anything, content_type: 'image/jpeg',
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.